common/devdoc/sm_requirements.md

```diff
@@ -179,10 +182,26 @@ MOCKABLE_FUNCTION(, void, sm_open_end, SM_HANDLE, sm, bool, success);
 
 **SRS_SM_02_041: [** If state is not `SM_OPENING` then `sm_open_end` shall return. **]**
 
+If `SM_FAULTED_BIT` is 1 then `sm_open_end` shall return.
+
 **SRS_SM_02_074: [** If `success` is `true` then `sm_open_end` shall switch the state to `SM_OPENED`. **]**
 
 **SRS_SM_02_075: [** If `success` is `false` then `sm_open_end` shall switch the state to `SM_CREATED`. **]**
 
+### sm_on_open_complete
+
+```c
+MOCKABLE_FUNCTION(, SM_RESULT, sm_on_open_complete, SM_HANDLE, sm);
+```
+
+`sm_on_open_complete` informs the user that calling on_open_complete is allowed.
```

Comment:
Not sure I understand this. How does it "inform the user"?

I guess this is really "sm_is_in_opening" or something because it's checking the current state to make sure it's in OPENING.

But I don't understand how this helps because it's not synchronized with anything else. I assume the intention is to synchronize with `sm_on_error`. But it's entirely possible that this is called and returns GRANTED, then `sm_on_error` is called immediately after.

I think the only way for this to work is for it to behave like `sm_open_end` except it returns GRANTED or REFUSED. If you have the state change here, then it can synchronize with failures.

Similarly, you wouldn't need `sm_on_error` because that's just `sm_open_end` with success set to false.

Reply:
Ok, after a long spirited debate, we decided to go a different more difficult direction.